pkg/cli/client.go

```diff
@@ -19,8 +22,16 @@ import (
 )
 
 var httpClient *http.Client //nolint: gochecknoglobals
-
-const httpTimeout = 5 * time.Minute
+var mux sync.Mutex          //nolint: gochecknoglobals
```

Comment:
Please rename the variable to suggest what it's used for.

@rchincha @shimish2 we need this because the client transport is updated with certificates (which come from the http.Request object), and each request could have different host destinations, each needing a different CA or client certificate.

This does mean that all our requests would be done in a series.

Was there a specific reason why http.Client is a global? If it were local to doHTTPRequest we wouldn't have this problem.

Reply:
So we are going to be holding that mutex for the entire time we do a http request/response? Doesn't look right!

Reply:
Well yes, unless we make the assumption the client will connect to the same server every time and reuse the transport object with the certificates set the first time.

Or we make the client a local variable.

Reply:
I think setting the certificates once would be clean in: https://github.com/anuvu/zot/pull/209/files#diff-6661d8b960c79657b52bd4fd22680ca8b3608faa4d91c4e10ad52db3e6bf4300R36

Reply:
Why we need lock here, is it because of http client ?

Reply:
There's a data race here https://github.com/anuvu/zot/pull/209/files#diff-6661d8b960c79657b52bd4fd22680ca8b3608faa4d91c4e10ad52db3e6bf4300R129, and tests are failing.